unittest/client/src/ilm_control_unittests.cpp

```diff
@@ -0,0 +1,5071 @@
+ *                      -# Verification point:
+ *                         +# ilm_surfaceRemoveNotification() must return ILM_FAILED
+ *                         +# wl_proxy_marshal_flags() not be called
+ *                         +# wl_display_roundtrip_queue() must be called once time
+ */
+TEST_F(IlmControlTest, ilm_surfaceRemoveNotification_invalidSurface)
+{
+    ilm_context.initialized = true;
+
+    t_ilm_surface l_surface = 6;
+    ASSERT_EQ(ILM_FAILED, ilm_surfaceRemoveNotification(l_surface));
+
+    ASSERT_EQ(0, wl_proxy_marshal_flags_fake.call_count);
+    ASSERT_EQ(1, wl_display_roundtrip_queue_fake.call_count);
+}
+
+/** ================================================================================================
+ * @test_id             ilm_surfaceRemoveNotification_success
+ * @brief               Test case of ilm_surfaceRemoveNotification() where callback is not null pointer
+ * @test_procedure Steps:
+ *                      -# Set the ilm context initialized is true, ready init
+ *                      -# Calling the ilm_surfaceAddNotification() to add callback notification
+ *                      -# Calling the ilm_surfaceRemoveNotification()
+ *                      -# Verification point:
+ *                         +# ilm_surfaceRemoveNotification() must return ILM_SUCCESS
+ *                         +# wl_proxy_marshal_flags() must be called 2 times
+ *                         +# wl_display_roundtrip_queue() must be called 4 times
+ */
+TEST_F(IlmControlTest, ilm_surfaceRemoveNotification_success)
+{
+    ilm_context.initialized = true;
+
+    t_ilm_surface l_surface = 1;
+    ASSERT_EQ(ILM_SUCCESS, ilm_surfaceAddNotification(l_surface, &surfaceCallbackFunction));
+    ASSERT_EQ(ILM_SUCCESS, ilm_surfaceRemoveNotification(l_surface));
+
+    ASSERT_EQ(2, wl_proxy_marshal_flags_fake.call_count);
+    ASSERT_EQ(4, wl_display_roundtrip_queue_fake.call_count);
+}
+
+/** ================================================================================================
+ * @test_id             ilm_registerNotification_cannotGetRoundTripQueue
+ * @brief               Test case of ilm_registerNotification() where wl_display_roundtrip_queue() fails, return -1
+ * @test_procedure Steps:
+ *                      -# Set the ilm context initialized is true, ready init
+ *                      -# Mocking the wl_display_roundtrip_queue() does return -1
+ *                      -# Calling the ilm_registerNotification()
+ *                      -# Verification point:
+ *                         +# ilm_registerNotification() must return ILM_FAILED
+ *                         +# wl_display_roundtrip_queue() must be called once time
+ */
+TEST_F(IlmControlTest, ilm_registerNotification_cannotGetRoundTripQueue)
+{
+    ilm_context.initialized = true;
+    SET_RETURN_SEQ(wl_display_roundtrip_queue, mp_failureResult, 1);
+
+    ASSERT_EQ(ILM_FAILED, ilm_registerNotification(&notificationCallback, nullptr));
```

Comment:
In cases where api makes a call to sync_and_acquire_instance we should check for call_count matches the release_instance call_count.

Reply:
Those functions are internal of `ilm_control`, we didn't mock them.